src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -468,10 +471,19 @@ private void exitPosition(Spread spread) {
     }
 
     // if feeComputation == CLIENT we want to compute the fees and add them to the volume
-    private BigDecimal addFees(Exchange exchange, CurrencyPair currencyPair, BigDecimal volume) {
+    private BigDecimal addFees(Exchange exchange, CurrencyPair currencyPair, BigDecimal volume, boolean isShortFee) {
         if (exchangeService.getExchangeMetadata(exchange).getFeeComputation().equals(FeeComputation.CLIENT)) {
+            final ExchangeFee exchangeFee = exchangeService.getExchangeFee(exchange, currencyPair, true);
+
+            if (isShortFee && !exchangeFee.getMarginFee().isPresent()) {
+                LOGGER.error("exchange:{}|error while adding fees. Reason: missing margin fee for this exchange. Go to your application.yml and set a margin fee for this exchange",
+                    exchange.getExchangeSpecification().getExchangeName());
+                // Crash the bot
+                throw new RuntimeException("Missing margin fee configuration for exchange: " + exchange.getExchangeSpecification().getExchangeName());
+            }
+
             BigDecimal fee = volume
-                .multiply(exchangeService.getExchangeFee(exchange, currencyPair, true))
+                .multiply(isShortFee ? exchangeFee.getMarginFee().get() : exchangeFee.getTradeFee())
```

Comment:
Again if it's a short trade the trade fee and margin fee need to be added together.